lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -370,54 +425,97 @@ def match_annotations(annotations)
       result
     end
 
-    def start_watch
+    def start_pod_watch
       begin
         resource_version = @client.get_pods.resourceVersion
         watcher          = @client.watch_pods(resource_version)
       rescue Exception => e
-        message = "Exception encountered fetching metadata from Kubernetes API endpoint: #{e.message}"
+        message = "start_pod_watch: Exception encountered setting up pod watch from Kubernetes API #{@apiVersion} endpoint #{@kubernetes_url}: #{e.message}"
         if e.respond_to?(:response)
           message += " (#{e.response})"
         end
-
+        log.debug message
         raise Fluent::ConfigError, message
       end
 
       watcher.each do |notice|
         case notice.type
           when 'MODIFIED'
             cache_key = "#{notice.object['metadata']['namespace']}_#{notice.object['metadata']['name']}"
-            cached    = @cache[cache_key]
+            cached    = @pod_cache[cache_key]
             if cached
-              @cache[cache_key] = parse_pod_metadata(notice.object)
+              @pod_cache[cache_key] = parse_pod_metadata(notice.object)
+              @pod_cache_watch_updates += 1
+            else
+              @pod_cache_watch_misses += 1
             end
           when 'DELETED'
             cache_key = "#{notice.object['metadata']['namespace']}_#{notice.object['metadata']['name']}"
-            @cache.delete(cache_key)
+            @pod_cache.delete(cache_key)
+            @pod_cache_watch_deletes += 1
           else
             # Don't pay attention to creations, since the created pod may not
             # end up on this node.
+            @pod_cache_watch_ignored += 1
         end
       end
     end
 
     def start_namespace_watch
-      resource_version = @client.get_namespaces.resourceVersion
-      watcher          = @client.watch_namespaces(resource_version)
+      begin
+        resource_version = @client.get_namespaces.resourceVersion
+        watcher          = @client.watch_namespaces(resource_version)
+      rescue Exception => e
+        message = "start_namespace_watch: Exception encountered setting up namespace watch from Kubernetes API #{@apiVersion} endpoint #{@kubernetes_url}: #{e.message}"
+        if e.respond_to?(:response)
+          message += " (#{e.response})"
+        end
+        log.debug message
+        raise Fluent::ConfigError, message
+      end
+
       watcher.each do |notice|
-        puts notice
```

Comment:
Maybe this should be replaced with an info or debug message?

Reply:
I would say `trace` level - too verbose otherwise